python-manual/modules/ROOT/pages/query-simple.adoc

```diff
@@ -0,0 +1,251 @@
+
+[NOTE]
+There can be circumstances where your query structure prevents the usage of parameters in all its parts. For those advanced use cases, see xref:query-advanced#_dynamic_values_in_property_keys_relationship_types_and_labels[Dynamic values in property keys, relationship types, and labels].
+
+
+== Anatomy of a query
+
+Before you can run a query, you need to obtain a session from the driver with the `Driver.session()` method *(1)*. The `database` parameter is optional but recommended for performance, and further xref:transactions#_session_configuration[configuration parameters] can be included.
+
+Within a session, use the methods link:{neo4j-docs-base-uri}/api/python-driver/current/api.html#neo4j.Session.execute_read[`Session.execute_read()`] and link:{neo4j-docs-base-uri}/api/python-driver/current/api.html#neo4j.Session.execute_write[`Session.execute_write()`] *(2)*, depending on whether you want to retrieve data from the database or alter it. Both methods take a _transaction function_ callback *(3)* and an arbitrary number of positional and keyword arguments *(4)* which are handed down to the transaction function.
+
+The transaction function *(5)* is responsible for actually carrying out the queries and xref:result.adoc[processing the result]. Queries are specified with the link:{neo4j-docs-base-uri}/api/python-driver/current/api.html#neo4j.Transaction.run[`Transaction.run()`] method *(6)*, which returns a link:{neo4j-docs-base-uri}/api/python-driver/current/api.html#neo4j.Result[`Result`] object. You can then process the result *(7)* using xref:result#manipulate-result[any of the `Result` methods], or simply casting it to `list`.
+
+.Retrieve people whose name starts with `Al`.
+[source,python]
+----
+def match_person_nodes(tx, name_filter): # <5>
+    result = tx.run(""" # <6>
+        MATCH (p:Person) WHERE p.name STARTS WITH $filter
+        RETURN p.name AS name ORDER BY name
+        """, filter=name_filter)
+    return list(result)  # return a list of Record objects # <7>
+
+with driver.session(database="neo4j") as session:  # <1>
+    people = session.execute_read(  # <2>
+        match_person_nodes, # <3>
+        "Al", # <4>
+    )
+    for person in people:
+        print(person.data())  # obtain dict representation
+----
+
+You can find more information about sessions and transactions in the section xref:transactions.adoc[Run your own transactions].
+
+[TIP]
+The driver might automatically retry to run a failed transaction. For this reason, *transaction functions must be _idempotent_* (i.e., they should produce the same effect when run several times), because you do not know how many times they might be executed. In practice, this means that you should not edit nor rely on globals, for example. Note that although transactions functions might be executed multiple times, the queries inside it will always run only once.
```

Comment:
```suggestion
The driver might automatically retry to run a failed transaction. 
For this reason, *transaction functions must be _idempotent_* (i.e., they should produce the same effect when run several times), because you do not know how many times they might be executed. 
In practice, this means that you should not edit nor rely on globals, for example. 
Note that although transaction functions might be executed multiple times, the queries inside it are always run only once.
```